ballista/rust/executor/src/main.rs

```diff
@@ -94,32 +97,54 @@ async fn main() -> Result<()> {
             .clone()
             .map(executor_registration::OptionalHost::Host),
         port: port as u32,
+        grpc_port: grpc_port as u32,
     };
+    let executor_specification = ExecutorSpecification {
+        task_slots: opt.concurrent_tasks as u32,
+    };
+    let executor = Arc::new(Executor::new_with_specification(
+        &work_dir,
+        executor_specification,
+    ));
 
     let scheduler = SchedulerGrpcClient::connect(scheduler_url)
         .await
         .context("Could not connect to scheduler")?;
 
-    let executor = Arc::new(Executor::new(&work_dir));
-
-    let service = BallistaFlightService::new(executor.clone());
+    let scheduler_policy = opt.task_scheduling_policy;
+    match scheduler_policy {
+        TaskSchedulingPolicy::PushStaged => {
+            tokio::spawn(executor_server::startup(
+                scheduler,
+                executor.clone(),
+                executor_meta,
+            ));
+        }
+        _ => {
+            tokio::spawn(execution_loop::poll_loop(
```

Comment:
What's the reason for keeping the polling design around? I am a little bit concerned about the complexity to maintain these two models. From an end user's point of view, which one should be chosen under what circumstance?

Reply:
Hi @houqp, actually I prefer to only keep the push-based model. Maybe once the push-based model become stable, we can remove the pull-based one. Or do more refactoring to move the pull-based code to some place not so obvious.